components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2276,13 +2301,24 @@ private Group getGroupWithName(String groupName)
         }
 
         Group group = new Group();
-        group.setDisplayName(groupName);
+        if (mandateDomainForUsernamesAndGroupNamesInResponse()) {
+            groupName = prependPrimaryDomain(groupName);
+            group.setDisplayName(groupName);
+        } else if (mandateDomainForGroupNamesInGroupsResponse()) {
```

Comment:
In what kind of a scenario, this condition and the previous `if` condition differ?

Reply:
There was a case where primary domain is not depicted with users being returned at all but only on groups.We discussed to have a separate property to enable that behavior to ensure backward compatibility and to deprecate that in a later version. This is to check that separate property.